Library/Homebrew/hardware.rb

```diff
@@ -39,14 +71,17 @@ def family
       end
 
       def cores
-        1
+        return @cores if @cores
+        @cores = Utils.popen_read("getconf", "_NPROCESSORS_ONLN").chomp.to_i
+        @cores = 1 unless $CHILD_STATUS.success?
+        @cores
       end
 
       def bits
-        case RUBY_PLATFORM
-        when /x86_64/, /ppc64/ then 64
-        when /i\d86/, /ppc/ then 32
-        end
+        return @bits if @bits
+        @bits = Utils.popen_read("getconf", "LONG_BIT").chomp.to_i
+        raise ErrorDuringExecution.new("getconf", ["LONG_BIT"]) unless $CHILD_STATUS.success?
```

Comment:
Feels weird that this behaves differently to `cores` above; why should one raise an exception but not the other?

Reply:
With `cores` we have a reasonable default value that we can use, namely `1`. With `bits`, I'm not sure what a reasonable default value should be. I don't expect `getconf LONG_BIT` to fail though. Every system I've encountered has it. I was just being paranoid.